scripts/install.sh

```diff
@@ -0,0 +1,772 @@
+
+# If there's an existing installation in a different location, it shows a message and prompts the user to continue.
+
+# The resulting directory structure would be as follows:
+#
+# ── mathesar/ (parent installation directory - must exist)
+#    ├── bin/
+#    │   ├── mathesar                   (main executable bash script)
+#    │   ├── mathesar_path_source       (path setting script for sh-like shells)
+#    │   └── mathesar_path_source.fish  (path setting script for fish - if fish is installed)
+#    ├── uv/
+#    ├── packages/
+#    ├── mathesar-venv/
+#    ├── .env
+#    ├── .media
+#    ├── config/
+#    ├── db/
+#    ├── mathesar/
+#    └── ... (other source files)
+
+
+#=======CONFIGURATIONS=========================================================
+
+MATHESAR_VERSION="0.2.3"
+REQUIRED_UV_VERSION="0.6.13"
+
+VENV_DIR_NAME="mathesar-venv"
+ENV_FILE_NAME=".env"
+
+FORCE_DOWNLOAD_PYTHON=false
+CONNECTION_STRING=""
+NO_PROMPT=false
+PACKAGE_LOCATION="https://github.com/mathesar-foundation/mathesar/releases/download/${MATHESAR_VERSION}/mathesar.tar.gz"
+
+# This is called before installation
+set_install_dir() {
+  INSTALL_DIR="$(cd -P "$1" && pwd)"
+
+  UV_DIR="${INSTALL_DIR}"/uv
+  PACKAGE_DIR="${INSTALL_DIR}"/packages
+  PACKAGE_LOCAL_ARCHIVE="${PACKAGE_DIR}"/mathesar-"${MATHESAR_VERSION}".tar.gz
+
+  # Required by uv
+  export UV_UNMANAGED_INSTALL="${UV_DIR}"
+  export UV_CACHE_DIR="${UV_DIR}"/cache
+  export UV_PYTHON_INSTALL_DIR="${UV_DIR}"/python
+  export UV_PROJECT_ENVIRONMENT="${INSTALL_DIR}"/"${VENV_DIR_NAME}"
+}
+
+# Color definitions for output
+RED=$(tput setaf 1 2>/dev/null || echo "")
+GREEN=$(tput setaf 2 2>/dev/null || echo "")
+BLUE=$(tput setaf 4 2>/dev/null || echo "")
+RESET=$(tput sgr0 2>/dev/null || echo "")
+
+
+#=======COMMON UTILITIES=======================================================
+
+info() {
+  echo -e "${BLUE}==> $1${RESET}"
+}
+
+success() {
+  echo -e "${GREEN}==> $1${RESET}"
+}
+
+err_msgonly() {
+  echo -e "${RED}ERROR: $1${RESET}" >&2
+}
+
+err() {
+  err_msgonly "$1"
+  echo -e "${RED}Mathesar installation failed!${RESET}"
+  exit 1
+}
+
+command_exists() {
+  command -v "$1" > /dev/null 2>&1
+  return $?
+}
+
+require_command() {
+  if ! command_exists "$1"; then
+    err "The install script requires '$1' (command not found)"
+  fi
+}
+
+# For commands that should ideally never fail
+ensure() {
+  if ! "$@"; then err "Command failed: $*"; fi
+}
+
+run_cmd() {
+  # Execute the command, pipe output to indent each line with 4 spaces.
+  "$@" 2>&1 | sed 's/^/    /'
+  local status=${PIPESTATUS[0]}
+  if [[ $status -ne 0 ]]; then
+    err "Command failed: $*"
+  fi
+}
+
+download() {
+  if command_exists curl; then
+    info "Downloading from $1 (using curl)..."
+    run_cmd curl -sSfL "$1" -o "$2"
+  elif command_exists wget; then
+    info "Downloading from $1 (using wget)..."
+    run_cmd wget "$1" -O "$2"
+  else
+    require_command "curl or wget"
```

Comment:
I'd be shocked if a user doesn't have `wget` on a functioning *nix system. Also, wrapping in `run_cmd` here will distract in error cases (as above).

Reply:
See comment on error cases here: https://github.com/mathesar-foundation/mathesar/pull/4399#discussion_r2077379112